plugin/trino-memsql/src/test/java/io/trino/plugin/memsql/TestMemSqlClientModule.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.memsql;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static io.trino.testing.assertions.Assert.assertEquals;
+
+public class TestMemSqlClientModule
```

Comment:
`TestMemSqlClient`/`TestMemSqlPlugin` - IIRC we already have some tests which check the JDBC url in one of those classes.

Reply:
1:1 mapping between class versus test class (allows things like package private access to methods, etc).
If there is a document which expounds on Trino's testing methodology I'd love to read it 
( I've read https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md - it only offers stylistic guidelines, it doesn't tackle issues like this )

Reply:
Keeping this here because it tests the behaviour in 35ms, and is a great help to any developer who has to modify the behaviour.